### PR TITLE
V3.6.0 scene view native

### PR DESCRIPTION
**Pull request by gameall3d**
Re: cocos-creator/3d-tasks#

Changelog:
 * fix some problem when nativize scene editor

### CODE REVIEW

**Review by SantyWang — APPROVED** [review]

File: cocos/core/assets/asset.jsb.ts [hunk 1]
```diff
@@ -35,9 +35,14 @@ import { createMap } from '../utils/js-typed';
...
 import { getUrlWithUuid } from '../asset-manager/helper';
 import { extname } from '../utils/path';
-
+import { CCClass } from '../data/class';
+import { editorExtrasTag } from '../data/editor-extras-tag';
 declare const jsb: any;
 
+CCClass.fastDefine('jsb.CCObject', jsb.CCObject, { _name: '', _objFlags: 0, [editorExtrasTag]: {} });
+CCClass.Attr.setClassAttr(jsb.CCObject, editorExtrasTag, 'editorOnly', true);
+CCClass.Attr.setClassAttr(jsb.CCObject, 'replicated', 'visible', false);
```
Comment (pandamicro) [inline cocos/core/assets/asset.jsb.ts]: Move to object.jsb.ts

File: cocos/core/assets/asset.jsb.ts [hunk 2]
```diff
@@ -35,9 +35,14 @@ import { createMap } from '../utils/js-typed';
 import { property } from '../data/class-decorator';
 import { getUrlWithUuid } from '../asset-manager/helper';
 import { extname } from '../utils/path';
-
+import { CCClass } from '../data/class';
+import { editorExtrasTag } from '../data/editor-extras-tag';
 declare const jsb: any;
 
+CCClass.fastDefine('jsb.CCObject', jsb.CCObject, { _name: '', _objFlags: 0, [editorExtrasTag]: {} });
```
Comment (shrinktofit) [inline cocos/core/assets/asset.jsb.ts]: This piece of code should be sync to all portion that reference the `CCObject`. That's, you should define `CCObject.jsb.ts` and mapping it in `cc.config.json`.

File: cocos/core/scene-graph/base-node.jsb.ts [hunk 3]
```diff
@@ -27,14 +27,16 @@ import { ccclass, editable, serializable } from 'cc.decorator';
...
+import { baseNodePolyfill } from './base-node-dev';
+const baseNodeProto: any = jsb.BaseNode.prototype;
 
 baseNodeProto.createNode = null!;
 
 export type BaseNode = jsb.BaseNode;
 export const BaseNode = jsb.BaseNode;
 
+
+
```
Comment (pandamicro) [inline cocos/core/scene-graph/base-node.jsb.ts]: Useless blank lines

**Review by pandamicro — CHANGES_REQUESTED** [review]

File: cocos/core/assets/asset.jsb.ts (hunk 1 above)
Comment (gameall3d) [reply cocos/core/assets/asset.jsb.ts]: move to object.ts with ‘JSB’ check

**Review by shrinktofit — APPROVED** [review]

**Review by pandamicro — APPROVED** [review]

**Pull request merged**